### PR TITLE
Docs for SQLite state store

### LINKED ISSUE #3085: [1.10] Docs for SQLite state store
Need docs for the SQLite state store component: https://github.com/dapr/components-contrib/pull/2465

Some initial work on the prior pluggable component: https://github.com/ItalyPaleAle/dapr-sqlite-statestore

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes #3085

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+| `timeoutInSeconds` | N | Timeout, in seconds, for all database operations. Defaults to `20` | `30`
+| `tableName` | N | Name of the table where the data is stored. Defaults to `state`. | `"state"`
+| `cleanupIntervalInSeconds` | N | Interval, in seconds, to clean up rows with an expired TTL. Default: `3600` (i.e. 1 hour). Setting this to values <=0 disables the periodic cleanup. | `1800`, `-1`
+| `actorStateStore` | N | Consider this state store for actors. Defaults to `"false"` | `"true"`, `"false"`
+
+The **`connectionString`** parameter configures how to open the SQLite database.
+
+- Normally, this is the path to a file on disk, relative or absolute. For example: `"data.db"` or `"/mnt/data/mydata.db"`.
+- The path is interpreted by the SQLite library, so it's possible to pass additional options to the SQLite driver using "URI options". For example, `"path/to/data.db?mode=ro"` opens the database at path `path/to/data.db` in read-only mode. Please refer to the SQLite documentation for all supported [URI options](https://www.sqlite.org/uri.html).
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]:
```suggestion
- The path is interpreted by the SQLite library, so it's possible to pass additional options to the SQLite driver using "URI options". For example, `"path/to/data.db?mode=ro"` opens the database at path `path/to/data.db` in read-only mode. Refer to the SQLite documentation for all supported [URI options](https://www.sqlite.org/uri.html).
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+|--------------------|:--------:|---------|---------|
+| `connectionString` | Y | The connection string for the SQLite database. See below for more details. | `"path/to/data.db"`, `":memory:"`
+| `timeoutInSeconds` | N | Timeout, in seconds, for all database operations. Defaults to `20` | `30`
+| `tableName` | N | Name of the table where the data is stored. Defaults to `state`. | `"state"`
+| `cleanupIntervalInSeconds` | N | Interval, in seconds, to clean up rows with an expired TTL. Default: `3600` (i.e. 1 hour). Setting this to values <=0 disables the periodic cleanup. | `1800`, `-1`
+| `actorStateStore` | N | Consider this state store for actors. Defaults to `"false"` | `"true"`, `"false"`
+
+The **`connectionString`** parameter configures how to open the SQLite database.
+
+- Normally, this is the path to a file on disk, relative or absolute. For example: `"data.db"` or `"/mnt/data/mydata.db"`.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]: It is always good to be clear that a path is relative to what? Relative to the location of the component manifest? 

I would add an example of saying that, if you are using local development then you can place the sqllite.yaml file and the data.db file into the default components folder https://docs.dapr.io/getting-started/install-dapr-selfhost/#step-5-verify-components-directory-has-been-initialized

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+- Longer intervals require less frequent scans for expired rows, but can require storing expired records for longer, potentially requiring more storage space. If you plan to store many records in your state table, with short TTLs, consider setting `cleanupIntervalInSeconds` to a smaller value, for example `300` (300 seconds, or 5 minutes).
+- If you do not plan to use TTLs with Dapr and the SQLite state store, you should consider setting `cleanupIntervalInSeconds` to a value <= 0 (e.g. `0` or `-1`) to disable the periodic cleanup and reduce the load on the database.
+
+The column in the state table where the expiration date for records is stored in, `expiration_time`, **does not have an index by default**, so each periodic cleanup must perform a full-table scan. If you have a table with a very large number of records, and only some of them use a TTL, you may find it useful to create an index on that column. Assuming that your state table name is `state` (the default), you can use this query:
+
+```sql
+CREATE INDEX idx_expiration_time
+  ON state (expiration_time);
+```
+
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md:81]: Worth adding any instructions on installing SQLLite locally or using a containerized version on K8s? For example having a Setup section like this https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-memcached/#setup-memcached

Maybe at least say that you can create a DB file in the local development folder, given this DB is so easy to use (especially on a Mac)

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md (hunk 2 above)
Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]: It’s actually relative to the current working directory, that’s what we can say. Definitely not the path of the component manifest (very unlikely at least).

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md (hunk 3 above)
Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md:81]: No need to install SQLite :) The great thing about SQLite (and why it’s the most widely-used DB in the world by a large margin - literally [over 1 trillion](https://sqlite.org/index.html) of them) is that it’s entirely contained within the application. No external service or process to install or start.

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md (hunk 2 above)
Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]: The working directory for the executable? Yes I would add this and an example where possible. And if the component is loaded and the database is not found, is there a helpful error message telling you this and how to fix this?

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]: If the component is loaded and the database is not found, a new DB is created. That's default SQLite behavior.

I'll update to specify it's the CWD.

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]: Updated

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  - "/operations/components/setup-state-store/supported-state-stores/setup-sqlite/"
+---
+
+This component allows using SQLite 3 as state store for Dapr.
+
+> The component is currently compiled with SQLite version 3.40.1.
+
+## Create a Dapr component
+
+Create a file called `sqlite.yaml`, paste the following and replace the `<CONNECTION STRING>` value with your connection string, which is the path to a file on disk.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]:
```suggestion
Create a file called `sqlite.yaml`, paste the following, and replace the `<CONNECTION STRING>` value with your connection string, which is the path to a file on disk.
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+| `timeoutInSeconds` | N | Timeout, in seconds, for all database operations. Defaults to `20` | `30`
+| `tableName` | N | Name of the table where the data is stored. Defaults to `state`. | `"state"`
+| `cleanupIntervalInSeconds` | N | Interval, in seconds, to clean up rows with an expired TTL. Default: `3600` (i.e. 1 hour). Setting this to values <=0 disables the periodic cleanup. | `1800`, `-1`
+| `actorStateStore` | N | Consider this state store for actors. Defaults to `"false"` | `"true"`, `"false"`
+
+The **`connectionString`** parameter configures how to open the SQLite database.
+
+- Normally, this is the path to a file on disk, relative to the current working directory, or absolute. For example: `"data.db"` (relative to the working directory) or `"/mnt/data/mydata.db"`.
+- The path is interpreted by the SQLite library, so it's possible to pass additional options to the SQLite driver using "URI options". For example, `"path/to/data.db?mode=ro"` opens the database at path `path/to/data.db` in read-only mode. Refer to the SQLite documentation for all supported [URI options](https://www.sqlite.org/uri.html).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]:
```suggestion
- The path is interpreted by the SQLite library, so it's possible to pass additional options to the SQLite driver using "URI options". For example: `"path/to/data.db?mode=ro"` opens the database at path `path/to/data.db` in read-only mode. [Refer to the SQLite documentation for all supported URI options](https://www.sqlite.org/uri.html).
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+- The special case `":memory:"` launches the component backed by an in-memory SQLite database. This database is not persisted on disk, not shared across multiple Dapr instances, and all data is lost when the Dapr sidecar is stopped.
+
+## Advanced
+
+### TTLs and cleanups
+
+This state store supports [Time-To-Live (TTL)](https://docs.dapr.io/developing-applications/building-blocks/state-management/state-store-ttl/) for records stored with Dapr. When storing data using Dapr, you can set the `ttlInSeconds` metadata property to indicate after how many seconds the data should be considered "expired".
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]:
```suggestion
This state store supports [Time-To-Live (TTL)]({{< ref state-store-ttl.md >}}) for records stored with Dapr. When storing data using Dapr, you can set the `ttlInSeconds` metadata property to indicate when the data should be considered "expired".
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+Because SQLite doesn't have built-in support for TTLs, this is implemented in Dapr by adding a column in the state table indicating when the data is to be considered "expired". Records that are "expired" are not returned to the caller, even if they're still physically stored in the database. A background "garbage collector" periodically scans the state table for expired rows and deletes them.
+
+The interval at which the deletion of expired records happens is set with the `cleanupIntervalInSeconds` metadata property, which defaults to 3600 seconds (that is, 1 hour).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]:
```suggestion
The `cleanupIntervalInSeconds` metadata property sets the expired records deletion interval, which defaults to 3600 seconds (that is, 1 hour).
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+- Longer intervals require less frequent scans for expired rows, but can require storing expired records for longer, potentially requiring more storage space. If you plan to store many records in your state table, with short TTLs, consider setting `cleanupIntervalInSeconds` to a smaller value, for example `300` (300 seconds, or 5 minutes).
+- If you do not plan to use TTLs with Dapr and the SQLite state store, you should consider setting `cleanupIntervalInSeconds` to a value <= 0 (e.g. `0` or `-1`) to disable the periodic cleanup and reduce the load on the database.
+
+The column in the state table where the expiration date for records is stored in, `expiration_time`, **does not have an index by default**, so each periodic cleanup must perform a full-table scan. If you have a table with a very large number of records, and only some of them use a TTL, you may find it useful to create an index on that column. Assuming that your state table name is `state` (the default), you can use this query:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-sqlite.md]:
```suggestion
The `expiration_time` column in the state table, where the expiration date for records is stored, **does not have an index by default**, so each periodic cleanup must perform a full-table scan. If you have a table with a very large number of records, and only some of them use a TTL, you may find it useful to create an index on that column. Assuming that your state table name is `state` (the default), you can use this query:
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]

**Review by hhunter-ms — APPROVED** [review]

**Comment by ItalyPaleAle** [comment]
I've made one additional change to correct information about using URI options

**Pull request merged**